mcl-kzg/kzg/src/kzg_settings.rs

```diff
@@ -23,6 +25,85 @@ impl KZGSettings {
             secret2: sec2
         }
     }
+
+    pub fn check_proof_single(&self, commitment: &G1, proof: &G1, x: &Fr, y: &Fr) -> bool {
+        let g1_gen = G1::gen();
+        let g2_gen = G2::gen();
+        let secret_minus_x = &self.secret2[1] - &(&g2_gen * x); // g2 * x to get x on g2
+        let commitment_minus_y = commitment - &(&g1_gen * y);
+
+        Curve::verify_pairing(&commitment_minus_y, &g2_gen, proof, &secret_minus_x)
+    }
+
+    pub fn compute_proof_multi(&self, p: &Polynomial, x0: &Fr, n: usize) -> G1 {
+
+        let mut divisor: Polynomial = Polynomial { coeffs: Vec::new() };
+        let x_pow_n = x0.pow(n);
+        divisor.coeffs.push(x_pow_n.get_neg());
+
+        for _ in 1..n {
+            divisor.coeffs.push(Fr::zero());
+        }
+
+        divisor.coeffs.push(Fr::one());
+        let temp_poly = p.clone();
+        let q = temp_poly.div(&divisor.coeffs).unwrap();
+        let ret = q.commit(&self.secret1);
+
+        ret
+    }
+
+    pub fn check_proof_multi(&self, commitment: &G1, proof: &G1, x: &Fr, ys: &Vec<Fr>, n: usize) -> bool {
+        let mut interp = Polynomial::new(n);
+        interp.coeffs = self.fs.fft_from_slice(ys, true);
+
+        let inv_x = x.inverse();
+        let mut inv_x_pow = inv_x.clone();
+        for i in 1..n {
+            let mut temp_fr = Fr::zero();
+            Fr::mul(&mut temp_fr, &interp.coeffs[i], &inv_x_pow);
+            interp.coeffs[i] = temp_fr;
+
+            let mut temp_fr2 = Fr::zero();
+            Fr::mul(&mut temp_fr2, &inv_x_pow, &inv_x);
+            inv_x_pow = temp_fr2;
+        }
+
+        let x_pow = inv_x_pow.inverse();
+        let g2_gen = G2::gen(); 
+
+        let mut xn2 = G2::zero();
+        G2::mul(&mut xn2, &g2_gen, &x_pow);
```

Comment:
@ciliakas this was used, since same logic is used in trait implementations. But maybe it would be better to move this logic from trait implementations to data types files, since there are some inside calls, and using this every time seems a bit tedious?